src/v/cluster/feature_table.h

```diff
@@ -297,6 +304,9 @@ class feature_table {
     // Waiting for a particular feature to be preparing
     waiter_queue<feature> _waiters_preparing;
 
+    // Currently loaded redpanda license details
+    std::optional<security::license> _license;
```

Comment:
The idea of this going here isn't for optimization its just really because the feature table is most likely the best home for this. As for it existing on all shards, this is true however the same could be said about all of the member variables within feature table. I would defer this answer to @jcsp the root of the question being how come feature table was not constructed on a single shard vs all shards?

Reply:
I guess my question is already answered here. If we are going to put this in feature table, it will be on all shards. I think to the question of why feature table exists on all shards, there is a comment in the code for it.

https://github.com/redpanda-data/redpanda/commit/7118eb4382c05d86dee2fef181ac75475f0bf236#diff-ba9c402c3d7b7250b276cfe1fcbf4102ce72355ebe2d61d0c95289aa50e8cba4R31